src/win/tty.c

```diff
@@ -803,6 +836,46 @@ void uv_process_tty_read_raw_req(uv_loop_t* loop, uv_tty_t* handle,
           prefix_len = 0;
         }
 
+        /* Handle a couple of special cases here that may not hit the standard
+         * get_vt100_fn_key path. These match the behavior of zsh under
+         * WSL/conhost. There is a little bit of duplicate handling here and
+         * in get_vt100_fn_key. This is needed because of slight differences
+         * in the way conhost generates key presses directly or over a PTY. */
+        if (KEV.uChar.UnicodeChar == '\t') {
+          if (KEV.dwControlKeyState &
+              (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED)) {
+            continue;
+          } else if (KEV.dwControlKeyState & SHIFT_PRESSED) {
+            /* Sort of a hack, but it works - set the prefix to "^[[" and change
+             * the character value to 'Z' to get a backtab. */
+            handle->tty.rd.last_key[0] = '\033';
+            handle->tty.rd.last_key[1] = '[';
+            prefix_len = 2;
+            KEV.uChar.UnicodeChar = 'Z';
+          }
+          /* Otherwise fall through - it's just a regular tab. */
+        } else if (KEV.uChar.UnicodeChar == '2' ||
+                   KEV.uChar.UnicodeChar == ' ') {
+          /* Ctrl-2 or Ctrl-Space generate a NUL. */
+          if (KEV.dwControlKeyState &
+              (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED)) {
+            handle->tty.rd.last_key[prefix_len] = 0;
+            handle->tty.rd.last_key_len = 1 + prefix_len;
+            handle->tty.rd.last_key_offset = 0;
+            continue;
+          }
+        } else if (KEV.uChar.UnicodeChar == 0x1F) {
+          /* The Ctrl-_ key is a weird one. In conpty mode, we will see a key
+           * with UnicodeChar == 0x1F, which is handled here. With a direct
+           * console connection however, the UnicodeChar value is 0. The problem
+           * then becomes that the VK code of the '/' key depends on the user's
+           * keyboard layout. */
+          handle->tty.rd.last_key[prefix_len] = 0x1F;
```

Comment:
This seems the same as the default fall-through, but I don't know this aspect of the code very well. What am I missing?

Reply:
This is indeed unnecessary, I'll take it out.

Reply:
@parkovski did you get a chance to update this?